app/assets/javascripts/discourse/tests/acceptance/composer-image-preview-test.js

```diff
@@ -340,4 +340,27 @@ acceptance("Composer - Image Preview", function (needs) {
       "alt text updated"
     );
   });
+
+  test("Image delete button", async function (assert) {
+    await visit("/");
+    await click("#create-topic");
+
+    let uploads = [
+      "![image_example_0|666x500](upload://q4iRxcuSAzfnbUaCsbjMXcGrpaK.jpeg)",
+      "![image_example_1|481x480](upload://p1ijebM2iyQcUswBffKwMny3gxu.jpeg)",
+    ];
+
+    await fillIn(".d-editor-input", uploads.join("\n"));
+
+    uploads[0] = ""; // delete the first image.
```

Comment:
This feels a little awkward... can we be a bit more specific and check below that the composer does not have `image_example_0` but does have `image_example_1`?

Reply:
Added two more test case to explicity test if composer has image_example_0 and image_example_1